ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/template2/Template2.kt

```diff
@@ -0,0 +1,160 @@
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(UIConstant.defaultButtonVerticalSpacing, Alignment.CenterVertically),
+    ) {
+        // TODO-PAYWALLS: Replace with correct image
+        val drawable = LocalContext.current.packageManager.getApplicationIcon(LocalContext.current.packageName)
+        Image(drawable.toBitmap(config = Bitmap.Config.ARGB_8888).asImageBitmap(), contentDescription = "")
+
+        val localizedConfig = state.paywallData.localizedConfig()
+        Text(
+            style = MaterialTheme.typography.displaySmall,
+            fontWeight = FontWeight.Black,
+            textAlign = TextAlign.Center,
+            text = localizedConfig?.title ?: "",
+            color = state.paywallData.getColors().text1Color,
+        )
+        Text(
+            style = MaterialTheme.typography.bodyLarge,
+            fontWeight = FontWeight.Bold,
+            textAlign = TextAlign.Center,
+            text = localizedConfig?.subtitle ?: "",
+            color = state.paywallData.getColors().text1Color,
+        )
+        state.packages.forEach { aPackage ->
+            SelectPackageButton(state, aPackage, viewModel)
+        }
+    }
+}
+
+@Composable
+private fun PurchaseButton(state: PaywallViewState.Template2, viewModel: PaywallViewModel) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = UIConstant.defaultHorizontalPadding),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center,
+    ) {
+        val activity = LocalContext.current.getActivity() ?: error("Error finding activity")
+        val colors = state.paywallData.getColors()
+        Button(
+            modifier = Modifier.fillMaxWidth(),
+            onClick = { viewModel.purchaseSelectedPackage(activity) },
+            colors = ButtonDefaults.buttonColors(
+                containerColor = colors.callToActionBackgroundColor,
+                contentColor = colors.callToActionForegroundColor,
+            ),
+        ) {
+            Text(text = state.paywallData.localizedConfig()?.callToAction ?: "")
+        }
+    }
+}
+
+@Composable
+private fun RestorePurchasesButton(viewModel: PaywallViewModel) {
+    TextButton(onClick = { viewModel.restorePurchases() }) {
+        Text(text = stringResource(id = R.string.restore_purchases))
+    }
+}
+
+@Composable
+private fun SelectPackageButton(
+    state: PaywallViewState.Template2,
+    aPackage: Package,
+    viewModel: PaywallViewModel,
+) {
+    val colors = state.paywallData.getColors()
+    val isSelected = aPackage == state.selectedPackage
+    val (background, textColor) = if (isSelected) {
+        colors.accent2Color to colors.accent1Color
+    } else {
+        // TODO-PAYWALLS: Find correct background unselected color
+        colors.backgroundColor to colors.text1Color
```

Comment:
I didn't find what color to use for the unselected package button... Seems like in iOS we just use the default? Or maybe I missed it @NachoSoto

Reply:
The logic is a bit complex, but TLDR it's no background colors:
```swift
if selected {
    self.roundedRectangle
        .foregroundColor(self.selectedBackgroundColor)
} else {
    if self.configuration.backgroundImageURLToDisplay != nil {
        // Blur background if there is a background image.
        self.roundedRectangle
            .foregroundStyle(.thinMaterial)
    } else {
        // Otherwise the text should have enough contrast with the selected background color.
        EmptyView()
    }
}
```

You can improve this in an upcoming PR, that's basically just polish :)